### ISSUE TITLE
Query with two `--exif` clauses seem to use OR instead of AND, resulting in double the lines in the output.

### ISSUE THREAD

**Issue by oPromessa** [RhetTbull/osxphotos#873]
**Before submitting a bug report, please ensure you are running the most recent version of osxphotos and that the bug is reproducible on the latest version**
* Done

**Describe the bug**
Based on [query documentation](https://rhettbull.github.io/osxphotos/cli.html#osxphotos-query) it is expected:
> (...) if more than one option is provided, they are treated as “AND” (e.g. search for photos matching all options).

But running the following two commands, selecting one and then two `--exif` TAG clauses, being two more restrictive, I thought it would perform the AND and result in an inferior number of results.

The thing is that it seems to be doubling the output and repeating UUIDs... possibly one for each of the `--exif clauses`, as they are both true for this particular case: Make is always `OLYMPUS (...)` and model is `uD800,S800`. **Third check** with `sort` an `uniq -c`.

```sh
$ osxphotos query  --edited --exif Model "uD800,S800" | wc -l 
2431
$ osxphotos query  --edited  --exif Make "OLYMPUS IMAGING CORP." --exif Model "uD800,S800" --field uuid "{uuid}" --field name "{original_name}" | wc -l 
4861
$ osxphotos query  --edited  --exif Make "OLYMPUS IMAGING CORP." --exif Model "uD800,S800" --field uuid "{uuid}" --field name "{original_name}" | sort | uniq -c 
   2 0019D6FC-C91F-4A10-BD21-6894FE2F96D6,P7120141
   2 0021379F-B868-4884-99FE-90ACC810288B,P5270399
   2 0022BD65-651D-4703-8DF5-D62921FBC24A,PC140128
   2 00297C4B-F015-4560-A8A5-C1042357D24E,P2100163
   2 005EF2FD-6073-47CB-83D6-0D03A44AA210,P5130059
(...)
```

**To Reproduce**
1. What' the full command line you used with osxphotos?
`osxphotos query  --edited  --exif Make "OLYMPUS IMAGING CORP." --exif Model "uD800,S800" --field uuid "{uuid}" --field name "{original_name}" | wc -l `
3. What was the error output?
No actual error. Doubled lines.

**Expected behavior**
An AND operator on both `--exif` entries and not returning double lines.

**Screenshots**
N/A

**Desktop (please complete the following information):**
 - OS: [e.g. which version of macOS]: **12.6.2**
 - osxphotos version (`osxphotos --version`): **0.55.2**

**Additional context**
N/A

**Comment by RhetTbull** [comment]
This is a documentation issue and a bug. The documentation issue is regarding the treatment of multiple options and AND/OR criteria. The bug is that the UUIDs are repeated in the query results when multiple `--exif` tags are provided. 

This is how query options work:

- If an option is repeated (e.g. `--uuid`, `--name`, etc.) then those are treated as "OR": find photos that match name X or match name Y. 
- If two different options are provided (e.g. `--name X --exif Make Canon`, these are treated as "AND": find photos matching name X and camera make "Canon"

I will update the documentation to try to make this more clear. It is stated so in documentation for many but not all of the options:

```
--name <FILENAME>
Search for photos with filename matching FILENAME. 
If more than one –name options is specified, they are treated as “OR”,
 e.g. find photos matching any FILENAME.
```

The rationale for this is that for many query options (name, uuid, etc.) a photo can match only one value. Thus providing multiple values is a way to find multiple photos. For common use cases where you are trying to select a specific subset of photos by matching different criteria, "AND" makes more sense.  The current design does not support any kind of boolean logic: e.g. `((X and Y) or Z))` and this would be difficult to implement from the command line. For these, a query function, written in python, would be more appropriate.  

I will also fix the bug so the that multiple `--exif` options returns the union of the results as do other options that are repeated.

For your use case, if you're trying to find where Make = Olympus AND Model = "uD800,S800", you could do this:

`--query-eval "photo.exif_info.camera_make == 'Olympus' and photo.exif_info.camera_model == 'uD800,S800'"`

If you want to find "uD800,S800" in the Model but not exact match, you could do this:

`--query-eval "photo.exif_info.camera_make == 'Olympus' and 'uD800,S800' in photo.exif_info.camera_model"`

**Comment by oPromessa** [comment]
Got it. Great!!! Thanks!

**Comment by RhetTbull** [comment]
This bug with duplicate results occurs for `--name` as well

**Comment by RhetTbull** [comment]
Fixed in [v0.55.3](https://github.com/RhetTbull/osxphotos/releases/tag/v0.55.3)
